lib/mk-flake.nix

```diff
@@ -140,63 +163,70 @@ let
     # pass along sops-nix overlay.
     ++ optionals withSops (singleton sops-nix.overlay);
 
-    # Evaluates to `packages.<system>.<pname> = <unstable-channel-reference>.<pname>`.
-    packagesBuilder = channels:
-      let
-        inherit (channels) nixpkgs;
-      in
-      recursiveUpdate
-        # these packages construct themselves if and only if the system is supported.
-        (import ../pkgs nixpkgs)
-        # pass along the packagesBuilder
-        (packagesBuilder channels);
-
-    # Evaluates to `devShell.<system> = "attributeValue"`
-    devShellBuilder = channels: with channels.nixpkgs;
-      let
-        _userShell = devShellBuilder channels;
-        userShell = if _userShell != null then _userShell else mkShell { };
-
-        # the base devShell.
-        baseShell = userShell.overrideAttrs (oa: {
-          name = "soxincfg";
-          buildInputs = (oa.buildInputs or [ ]) ++ [
+    # TODO: Add support for modifying the outputsBuilder.
+    outputsBuilder = channels: {
+      # Evaluates to `packages.<system>.coreutils = <unstable-nixpkgs-reference>.package-from-overlays`.
+      packages =
+        let
+          inherit (channels) nixpkgs;
+        in
+        recursiveUpdate
+          # these packages construct themselves if and only if the system is supported.
+          (import ../pkgs nixpkgs)
+          # pass along the packagesBuilder
+          (packagesBuilder channels);
+
+      # Evaluates to `devShell.<system> = <nixpkgs-channel-reference>.mkShell { name = "devShell"; };`.
+      devShell =
+        let
+          inherit (channels) nixpkgs;
+
+          inherit (nixpkgs)
+            mkShell
             nixpkgs-fmt
             pre-commit
+            sops
+            sops-pgp-hook
+            ssh-to-pgp
+            ;
+
+          _userShell = devShellBuilder channels;
+          userShell = if _userShell != null then _userShell else mkShell { name = "devShell"; };
+
+          # the base devShell.
+          baseShell = userShell.overrideAttrs (oa: {
+            name = "soxincfg";
+            buildInputs = (oa.buildInputs or [ ]) ++ [ nixpkgs-fmt pre-commit ];
+          });
+
+          # overlay the baseShell with things that are only necessary if the
+          # user has enabled sops support.
+          sopsShell = baseShell.overrideAttrs (oa: optionalAttrs withSops {
+            buildInputs = (oa.buildInputs or [ ]) ++ [ sops ssh-to-pgp ];
+            nativeBuildInputs = (oa.nativeBuildInputs or [ ]) ++ [ sops-pgp-hook ];
```

Comment:
Just a note, this is deprecated in latest sops versions, we should think about changing it.

Reply:
Do you have a link to the deprecation? What's the replacement? We should probably fix this in a separate PR.